src/components/TextInput/TextInput.tsx

```diff
@@ -192,6 +199,24 @@ export const TextInput = ({
                 data-test-id="text-input"
                 {...spellcheckProp}
             />
+            {extraAction && (
+                <button
+                    className={merge([
+                        'tw-flex tw-items-center tw-justify-center tw-transition-colors tw-rounded',
+                        disabled ? 'tw-cursor-default tw-text-black-40' : 'tw-text-black-60  hover:tw-text-black-100',
```

Comment:
Instead of `black-[40, 60, 100]` you should use Text Alias Tokens. You can find them here https://beta-fondue-components.frontify.com/?path=/story/tokens--alias-tokens